cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -91,46 +94,92 @@
     private final PersistedCassandraClusterHealthCheckHistory healthCheckHistory;
     @NotNull
     private final PersistedCassandraFrameworkConfiguration configuration;
+    @NotNull
+    private final PersistedCassandraClusterJobs jobsState;
+
+    private static CassandraCluster singleton;
+
+    public static CassandraCluster singleton() {
+        return singleton;
+    }
+
+    public static int getPortMapping(CassandraFrameworkConfiguration configuration, String name) {
+        for (PortMapping portMapping : configuration.getPortMappingList()) {
+            if (portMapping.getName().equals(name))
+                return portMapping.getPort();
+        }
+        Long port = defaultPortMappings.get(name);
+        if (port == null)
+            throw new IllegalStateException("no port mapping for " + name);
+        return port.intValue();
+    }
 
     public CassandraCluster(
         @NotNull final Clock clock,
         @NotNull final String httpServerBaseUrl,
-        @NotNull final ExecutorCounter execCounter,
-        @NotNull final PersistedCassandraClusterState clusterState,
-        @NotNull final PersistedCassandraClusterHealthCheckHistory healthCheckHistory,
+        @NotNull final State state,
+        @NotNull final ExecutorCounter executorCounter,
         @NotNull final PersistedCassandraFrameworkConfiguration configuration
     ) {
+        singleton = this;
         this.clock = clock;
         this.httpServerBaseUrl = httpServerBaseUrl;
-        this.execCounter = execCounter;
-        this.clusterState = clusterState;
-        this.healthCheckHistory = healthCheckHistory;
+        this.execCounter = executorCounter;
+        this.clusterState = new PersistedCassandraClusterState(state);
+        this.healthCheckHistory = new PersistedCassandraClusterHealthCheckHistory(state);
+        this.jobsState = new PersistedCassandraClusterJobs(state);
```

Comment:
Please revert this change. `CassandraCluster` should know nothing about how to instantiate the state objects.  If this is to facilitate testing the stateful objects should be instantated against either `org.apache.mesos.state.InMemoryState` or a mock. Remove `singleton`.

This class also shouldn't be used as a service locator.  If another class needs a common dependency with this class we should clean up the dependency tree so that things can be constructed with all dependencies.